test/simdloop.jl

```diff
@@ -174,3 +174,13 @@ Base.SimdLoop.simd_index(v::iter31113, j, i) = j
 Base.SimdLoop.simd_inner_length(v::iter31113, j) = 1
 Base.SimdLoop.simd_outer_range(v::iter31113) = v
 @test 2001000 == simd_sum_over_array(iter31113(Vector(1:2000)))
+
+# nested loops
+function nested_loops()
+    res = 0
+    @simd for i=1:2,j=1:2
```

Comment:
I think this looks misleading. Why not just write two loops and be explicit on which one the `@simd` is applied?

Reply:
Are you commenting on the test or the whole PR? The test is there to test the PR so needs to be this way.

Reply:
The whole PR. Commented here because this is how it will look when used.

Reply:
The main motivation is multidimensional array loops. You start writing things like
```
for k=2:N-1, j=2:N-1, i=2:N-1
    B[i,j,k] = A[i+1,j,k]-2A[i,j,k]+A[i-1,j,k]
end
```
or something, then you want to do simd on that and have to rewrite your nice loop to three loops, which is annoying.

Reply:
I am strongly in favour in having to be explicit which of the loops should get the `@simd` annotation.

Reply:
It seems reasonable for this to mean that all of the loops have simd permission, no?